primitives/forest-verifier/src/lib.rs

```diff
@@ -49,20 +54,7 @@ where
 
         // Check if the iterator has at least one leaf.
         if trie_de_iter.next().is_none() {
-            #[cfg(test)]
-            unreachable!(
-                "This should not happen. A trie with no leafs wouldn't be possible to create."
-            );
-
-            #[allow(unreachable_code)]
-            {
-                return Err("No leaves provided in proof.".into());
-            }
-        }
-
-        // Check that `challenges` is not empty.
-        if challenges.is_empty() {
-            return Err("No challenges provided.".into());
+            return Ok(BTreeSet::new());
```

Comment:
Why are you returning this? It doesn't make sense that this would be a valid way to pass a verification